docs/component_spec.md

```diff
@@ -0,0 +1,173 @@
+# Cytoxnet Component Specification
+The package is intended as an exploration and a toolbox for the prediction of
+cytotoxicity of compounds, but aims to leverage datasets of other targets
+for transfer learning.
+***
+***
+## package format
+```
+/cytoxnet
+|-/dataprep
+| |-io.py
+| |-featurize.py
+| |-dataprep.py
+| |-analyze.py
+|-/models
+| |-models.py
+| |-opt.py
+| |-analyze.py
+|-/tests
+```
+***
+
+## `cytoxnet/dataprep/`
+
+### `cytoxnet/dataprep/io.py`
+
+```
+load_file(datafile: str,
+          cols: str or list of str = None) -> pandas DataFrame
+```
+- __Use Case__: (1)
+- __Inputs__: File location string, columns to keep
+- __Outputs__: Dataframe of dataset on file
+
+```
+load_XXX(prepare: bool = False)
+```
+- __Use Case__: (1)
+- __Inputs__: Whether or not to featurize and prepare set using a default
+  pipeline
+- __Outputs__: Dataframe of package dataset
+
+### `cytoxnet/dataprep/featurize.py`
+
+```
+clean_dataframe(dataframe,
+                cols: str or list of str = None,
+                duplicates: str = 'drop',
+                nans: str = 'drop') -> pandas DataFrame
+```
+- __Use Case__: (1, 2)
+- __Inputs__: Dataframe, columns to clean, how to handle nans and duplicates.
+- __Outputs__: Dataframe
+
+```
+featurize(dataframe,
+          target_cols: list of str,
+          id_col: str = 'smiles',
```

Comment:
Would this also take and convert InChIs?  I'm not familiar enough with how compspecs should be formatted to know whether that would need to be explicitly stated here, so feel free to ignore this comment if I'm just being nit-picky

Reply:
yes, see above